src/db/sysdb_search.c

```diff
@@ -545,8 +550,19 @@ static char *enum_filter(TALLOC_CTX *mem_ctx,
         filter = talloc_asprintf(tmp_ctx, "(&%s", base_filter);
 
         if (filter != NULL && attr != NULL && attr_filter != NULL) {
-            filter = talloc_asprintf_append(filter, "(%s=%s)",
-                                            attr, attr_filter);
+            if (domain == NULL || !IS_SYSDB_NAME(attr)) {
+                value = attr_filter;
+            } else {
+                value = sss_create_internal_fqname(tmp_ctx, attr_filter, domain);
```

Comment:
(1) Looks like ldb cache indeed always store FQDN, but I wonder if it's documented anywhere...

(2) What is the guarantee that `attr_filter` is a short name here?

Reply:
(1) I really can't tell.

(2) None, but my understanding is that users should not be asking for the FQN as this is something internal to the cache; and if they do, the request will match no entry, which is correct.

Please correct me if I'm wrong.

Reply:
@pbrezina, is (1) correct?

Reply:
> my understanding is that users should not be asking for the FQN as this is something internal to the cache

You mean IFP d-bus API specifically?

I think in general that's ok and we do allow FQDN names as input in many interfaces. But probably cache_req engine will split it and keep short name here?

Reply:
It seems to be the case: https://github.com/SSSD/sssd/pull/5663

BTW, now I remember that every time I tried to use an `@` d-bus generated an error. A quick search showed that `@`, '`&` and `^` seem to be reserved for internal use.

```
# dbus-send --print-reply --system --dest=org.freedesktop.sssd.infopipe /org/freedesktop/sssd/infopipe/Users org.freedesktop.sssd.infopipe.Users.ListByName "string:scarter@*" "uint32:0"
Error sbus.Error.Errno: 1432158243: Cannot parse input
```

Reply:
> It seems to be the case: #5663

Ok.

> BTW, now I remember that every time I tried to use an `@` d-bus generated an error. A quick search showed that `@`, '`&` and `^` seem to be reserved for internal use.

That's weird user can't specify domain...

Reply:
> @pbrezina, is (1) correct?

AFAIK yes, name is always fqdn.